craft_parts/plugins/rust_plugin.py

```diff
@@ -87,7 +87,7 @@ def validate_environment(
         """
         if "rust-deps" in (part_dependencies or []):
             options = cast(RustPluginProperties, self._options)
-            if options.rust_channel:
+            if options.rust_channel and options.rust_channel != "none":
```

Comment:
Based on [the docs](https://canonical-craft-parts.readthedocs-hosted.com/en/latest/reference/plugins/rust_plugin.html), we can set `rust_channel` on `RustPluginProperties` to be `rust_channel: str = "stable"`, right? Then we only need `if options.rust_channel != "none":`

Reply:
This is for backward compatibility. For older `core20` Snaps, setting it as `stable` by default will break the build.

Reply:
Ahh, got it. Thank you!